zinc/src/main/scala/sbt/internal/inc/IncrementalCompilerImpl.scala

```diff
@@ -8,17 +8,18 @@ import xsbti.compile.{ CompileOrder, GlobalsCache, IncOptions, MiniSetup, Compil
 import xsbti.compile.{ PreviousResult, Setup, Inputs, IncrementalCompiler, PerClasspathEntryLookup }
 import xsbti.compile.{ Compilers, CompileProgress, JavaCompiler, JavaTools => XJavaTools, Output, ScalaCompiler, ClasspathOptions => XClasspathOptions }
 import java.io.File
-import sbt.util.Logger.m2o
+import sbt.util.Logger.{ m2o, f0 }
 import sbt.io.{ IO, Using }
 import xsbti.compile.CompileOrder.Mixed
+import sbt._
 
 // TODO -
 //  1. Move analyzingCompile from MixedAnalyzingCompiler into here
 //  2. Create AnalyzingJavaComiler class
 //  3. MixedAnalyzingCompiler should just provide the raw 'compile' method used in incremental compiler (and
 //     by this class.
 
-class IncrementalCompilerImpl extends IncrementalCompiler {
+class IncrementalCompilerImpl(externalLookup: Option[ExternalLookup] = None) extends IncrementalCompiler {
```

Comment:
I think ExternalLookup should be added to `xsbti.compile.Setup` like other setup related things. See my half done pull request https://github.com/sbt/zinc/pull/131.